src/curvefit/solvers/solvers.py

```diff
@@ -52,7 +96,7 @@ def fit(self, data, x_init=None, options=None):
             x0=x_init,
             jac=lambda x: self.model.gradient(x, data),
             bounds=self.model.bounds,
-            options=options,
+            options=self.options.to_dict(solver_type ="sciopt") if self.options is not None else None,
```

Comment:
do you need the `if else` since it will already been `None`?

Reply:
Yes, because otherwise when it tries to evaluate `self.options.to_dict(..)` when `options` is `None` it will crash.

Reply:
In this case it first checks that `self.options` is not `None` and only after that evaluates `self.options.to_dict(..) `

Reply:
got it!